### PR TITLE
Bcm2xxx extend mailbox

### PR DESCRIPTION
**Pull request by lulu98**
Extends the mailbox for the bcm2837 (rpi3) and bcm2711 (rpi4) with additional functionality.

### CODE REVIEW

File: libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h [hunk 1]
```diff
@@ -41,47 +41,77 @@ enum {
...
+
+////////////////////////////////////////////////////////////////////////////////
+//-----------------------------Mailbox Functions------------------------------//
+////////////////////////////////////////////////////////////////////////////////
+
 bool bcm2711_set_power_state_on(mailbox_t *mbox, uint32_t device_id);
 
 int bcm2711_get_clock_rate(mailbox_t *mbox, uint32_t clock_id);
 
+bool bcm2711_get_mac_address(mailbox_t *mbox, uint8_t buffer[6]);
```
Comment (axel-h) [inline libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h:115]:
```suggestion
bool bcm2711_get_mac_address(mailbox_t *mbox, uint8_t buffer[MAC_ADDRESS_SIZE]);
```

File: libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h [hunk 2]
```diff
@@ -41,47 +41,77 @@ enum {
...
+// See: https://github.com/raspberrypi/firmware/wiki/Mailbox-property-interface#get-board-mac-address
+#define MAC_ADDRESS_SIZE 6
+typedef struct {
+    MailboxInterface_PropertyTag_t tag;
+}
+PropertyTag_GetMACAddress_Request_t;
+
+typedef struct {
+    MailboxInterface_PropertyTag_t tag;
+    uint8_t mac_address[6];
```
Comment (axel-h) [inline libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h:101]:
```suggestion
    uint8_t mac_address[MAC_ADDRESS_SIZE];
```

File: libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h [hunk 3]
```diff
@@ -41,47 +41,77 @@ enum {
...
+
+////////////////////////////////////////////////////////////////////////////////
+//-----------------------------Mailbox Functions------------------------------//
+////////////////////////////////////////////////////////////////////////////////
+
 bool bcm2837_set_power_state_on(mailbox_t *mbox, uint32_t device_id);
 
 int bcm2837_get_clock_rate(mailbox_t *mbox, uint32_t clock_id);
 
+bool bcm2837_get_mac_address(mailbox_t *mbox, uint8_t buffer[6]);
```
Comment (axel-h) [inline libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h:115]:
```suggestion
bool bcm2837_get_mac_address(mailbox_t *mbox, uint8_t buffer[MAC_ADDRESS_SIZE]);
```

File: libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h [hunk 4]
```diff
@@ -41,47 +41,77 @@ enum {
...
+// See: https://github.com/raspberrypi/firmware/wiki/Mailbox-property-interface#get-board-mac-address
+#define MAC_ADDRESS_SIZE 6
+typedef struct {
+    MailboxInterface_PropertyTag_t tag;
+}
+PropertyTag_GetMACAddress_Request_t;
+
+typedef struct {
+    MailboxInterface_PropertyTag_t tag;
+    uint8_t mac_address[6];
```
Comment (axel-h) [inline libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h:101]:
```suggestion
    uint8_t mac_address[MAC_ADDRESS_SIZE];
```

**Comment by axel-h** [comment]
Looks ok to be, please address the comment with using the constatant.

**Review by axel-h — APPROVED** [review]

File: libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h (hunk 4 above)
Comment (lulu98) [reply libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h]: fixed

File: libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h (hunk 3 above)
Comment (lulu98) [reply libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h]: fixed

File: libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h (hunk 2 above)
Comment (lulu98) [reply libplatsupport/plat_include/bcm2837/platsupport/plat/mailbox_util.h]: fixed

File: libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h (hunk 1 above)
Comment (lulu98) [reply libplatsupport/plat_include/bcm2711/platsupport/plat/mailbox_util.h]: fixed

**Review by lsf37 — APPROVED** [review]

**Pull request merged**